### PR TITLE
feat(collapsible-sidebar): add prop to hide Box logo

### PR DESCRIPTION
**Pull request by Machoper**
Added a new prop `isLogoVisible` to optionally hide Box logo if set to false.

### CODE REVIEW

File: src/features/collapsible-sidebar/__tests__/CollapsibleSidebarLogo.test.js [hunk 1]
```diff
@@ -69,4 +69,13 @@ describe('components/core/collapsible-sidebar/CollapsibleSidebar', () => {
...
         expect(componentProp.find('LinkBase').prop('someprop')).toBe(someValue);
     });
+
+    test.each([true, false])('render Box logo when hideBoxLogo=%s', hideBoxLogo => {
+        const sidebar = getWrapper({
+            expanded: true,
+            hideBoxLogo,
+        });
+
+        expect(sidebar).toMatchSnapshot();
```
Comment (tjuanitas) [inline src/features/collapsible-sidebar/__tests__/CollapsibleSidebarLogo.test.js:79]: this snapshot looks expensive. are we able to just check for the existence of `bdl-CollapsibleSidebar-logoLink`?

File: src/features/collapsible-sidebar/CollapsibleSidebarLogo.js [hunk 2]
```diff
@@ -72,13 +72,15 @@ type Props = {
     /** Controls whether or not the sidebar is expanded on the page */
     expanded: boolean,
 
+    hideBoxLogo: boolean,
```
Comment (tjuanitas) [inline src/features/collapsible-sidebar/CollapsibleSidebarLogo.js:75]: can we do `isLogoVisible` and make it optional? we'd need to default it to `true`

File: src/features/collapsible-sidebar/__tests__/CollapsibleSidebarLogo.test.js (hunk 1 above)
Comment (Machoper) [reply src/features/collapsible-sidebar/__tests__/CollapsibleSidebarLogo.test.js]: Yes, I adjusted the tests.

File: src/features/collapsible-sidebar/CollapsibleSidebarLogo.js (hunk 2 above)
Comment (Machoper) [reply src/features/collapsible-sidebar/CollapsibleSidebarLogo.js]: Yes, I changed the prop name and set it to true by default.

**Review by tjuanitas — APPROVED** [review]

**Review by greathmaster — APPROVED** [review]

**Pull request merged**